### ISSUE TITLE
🐛[bug] 

### ISSUE THREAD

**Issue by xb478** [determined-ai/determined#6020]
### Describe the bug

Running a simple classification experiment with a learning rate scheduler defined as such:
`self.scheduler = self.context.wrap_lr_scheduler(
            torch.optim.lr_scheduler.CosineAnnealingLR(
                self.optimizer,
                T_max=self.context.get_hparam("t_max"),
                eta_min=self.context.get_hparam("eta_min"),
                verbose=False),
            step_mode=LRScheduler.StepMode.STEP_EVERY_BATCH)`

I assume that it should adjust the learning rate (calling self.scheduler.step()) after each batch.
Trial is configured as:
`hyperparameters:
   learning_rate: 1.0e-4
   learning_rate_decay: 1.0e-6
   global_batch_size: 32
   momentum: 0.9
   eta_min: 1.0e-8
   t_max: 20312
 records_per_epoch: 13000
 searcher:
   name: single
   metric: val_loss
   max_length:
      epochs: 50`

However, while in principle the learning rate is reduced with each batch there is also a stronger reduction happening at the last `scheduling_unit` of the Trial. `scheduling_unit` is left as default (`100` batches) and the last one is less than `100` which leads to the stronger reduction of the learning_rate. 

### Reproduction Steps

1. Define an experiment with a LearningRateScheduler
2. Setup scheduler with `LRScheduler.StepMode.STEP_EVERY_BATCH`
3. Leave `scheduling_unit=100` as default setting
4. Monitor learning rate

### Expected Behavior

Learning rate for schedulers should be only updated according to the chosen rule:
`LRScheduler.StepMode.STEP_EVERY_BATCH` no matter what `scheduling_unit` size is chosen to report training metrics.

### Screenshot

Current behavior as described:
![image](https://user-images.githubusercontent.com/70268572/219697325-5bdd130d-95ce-44a0-86b4-b34056755d11.png)

### Environment

- Device or hardware: <NA>
- OS: Ubuntu 22.04
- Version 0.19.9

**Comment by ioga** [comment]
thank you for the report, let us investigate.

**Comment by ioga** [comment]
I believe the LRScheduler steps are correct: you can check that by logging your `self.scheduler.get_last_lr()` in every batch.

However when we report and plot them, the metric value is averaged across the batches within the scheduling unit. That is, the LR number on the chart for batch #100 is not LR at batch 100 but average of LRs for batches 0 to 100, and this causes the plot to not have the smooth shape you are expecting.

In other words, the `scheduling_unit` selection here should not affect the training, but it'll affect how the plot looks.

**Comment by ioga** [comment]
will close this for now, please let us know if you need more help.
